study1/week6/decode-ways.java

```diff
@@ -0,0 +1,37 @@
+// https://leetcode.com/problems/decode-ways/
+// Time Complexity: O(n), n: s.length()
+// Space Complexity: O(n), n: s.length()
+class Solution {
+    public int numDecodings(String s) {
+        if (s.length() == 0) {  // edge case
+            return 0;
+        }
+
+        int length = s.length()+1;
+        int[] cntArr = new int[length]; // using dynamic programming
+        cntArr[0] = 1;
+
+        for (int i = 1; i < length; ++i) {
+            char ch = s.charAt(i-1);
+            if (ch >= '1' && ch <= '9') {   // check for 1-9
+                cntArr[i] += cntArr[i-1];
+            }
+
+            // check for 10-26
+            if (i >= 2) {
```

Comment:
You can start i from 2 and remove this check of ` if ( i >= 2) {`. Then, you also need to precompute `cntArr[1] = ...` as well